### PR TITLE
Improve customizability of `Select` component

### PR DESCRIPTION
**Pull request by bgptr**
https://github.com/decred/decrediton/pull/3631 needs additional customization for the `Select` component.

This diff adds:
- creatable select functionality 
- async select functionality
- possibility to add custom styling

### CODE REVIEW

File: src/components/Select/Select.jsx [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
+            border: "0.5rem solid transparent",
+            borderColor:
+              "var(--select-anchor-color) transparent transparent transparent",
+            position: "absolute",
+            top: "-0.3rem",
+            right: "0.3rem",
+            ...(props.selectProps.menuIsOpen && {
+              top: "-0.9rem",
+              borderColor:
+                "transparent transparent var(--color-primary) transparent",
```
Comment (amass01) [inline src/components/Select/Select.jsx:30]: avoid using inline styling, use conditioned classes

File: src/components/Select/Select.jsx [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
-        className={classNames(
-          styles.arrowAnchor,
-          props.selectProps.menuIsOpen && styles.arrowAnchorOpen
-        )}
-      />
+        style={{
+          position: "relative",
+          height: "100%",
+          width: "2rem",
+        }}>
```
Comment (amass01) [inline src/components/Select/Select.jsx:15]: same here

**Review by amass01 — COMMENTED** [review]
left couple of inline comments

File: src/components/Select/Select.jsx (hunk 1 above)
Comment (bgptr) [reply src/components/Select/Select.jsx:30]: @amass01, thanks for your review. I'd prefer the conditioned classes too, but with them, the arrow disappears in decrediton. The divs didn't receive the class definitions. I tried to figure out why, but only this inline styling works. Now, I revisited it without result. Do you have any idea how to solve this?

Comment (amass01) [reply src/components/Select/Select.jsx]: hrmm i'm not sure, I guess it related `react-select`, as the divs are wrapped with 
`components.DropdownIndicator`. I would just leave a comment about the limitation 
we can't find a  solution or a workaround. 

I see in the documentation that there is another `classNamePrefix` that might work. iirc we already
use that one.
See: https://react-select.com/styles#using-classnames

File: rollup.config.js [hunk 3]
```diff
@@ -27,9 +27,7 @@ const config = {
     external(),
     postcss({
       extract: true,
-      modules: {
-        globalModulePaths: ["src/components/Select/styles.css"],
```
Comment (amass01) [inline rollup.config.js:31]: wondering why we had this here, as `src/components/Select/styles.css` file didn't exist before this commit 🤔

Comment (bgptr) [reply rollup.config.js:31]: It didn't exist before this commit, but it did exist _before this PR_. 
The `globalModulePaths` config was introduced by: https://github.com/decred/pi-ui/pull/227

Comment (amass01) [reply rollup.config.js:31]: oh, i see, makes sense.

it was imported then as plain .css as it included pure css classes prefixed with `customSelect` which isn't
the case anymore, glad it evolved to a css module. 👍🏽

**Review by amass01 — APPROVED** [review]
ACK

**Review by tiagoalvesdulce — APPROVED** [review]
LGTM

**Pull request merged**